nautobot/users/api/views.py

```diff
@@ -40,6 +43,43 @@ class GroupViewSet(ModelViewSet):
     filterset_class = filters.GroupFilterSet
 
 
+#
+# REST API tokens
+#
+
+
+class TokenViewSet(ModelViewSet):
+    queryset = RestrictedQuerySet(model=Token).prefetch_related("user")
+    serializer_class = serializers.TokenSerializer
+    filterset_class = filters.TokenFilterSet
+
+    def get_queryset(self):
+        """
+        Limit users to their own Tokens.
+        """
+        queryset = super().get_queryset()
+        # Workaround for schema generation (drf_yasg)
+        if getattr(self, "swagger_fake_view", False):
+            return queryset.none()
+        return queryset.filter(user=self.request.user)
+
+
+class TokenProvisionView(APIView):
+    """
+    Non-authenticated REST API endpoint via which a user may create a Token.
+    """
+
+    authentication_classes = [authentication.TokenProvisionAuthentication]
+    permission_classes = []
+
+    def post(self, request):
+        # Create a new Token for the User
+        token = Token.objects.create(user=request.user)
+        data = serializers.TokenSerializer(token, context={"request": request}).data
+
+        return Response(data, status=HTTP_201_CREATED)
```

Comment:
I just realized that this could be further streamlined and doesn't need its own views. This suggestion moves `TokenProvisionView` into a ViewSet method at `TokenViewSet.provision()`. 

Additionally, I'll provide another comment that means the changes to `nautobot.users.api.urls.urlpatterns` can be reverted, since this route will be automatically generated.

```suggestion

# This needs to go up top in import section.
from rest_framework.decorators import action


class TokenViewSet(ModelViewSet):
    queryset = RestrictedQuerySet(model=Token).prefetch_related("user")
    serializer_class = serializers.TokenSerializer
    filterset_class = filters.TokenFilterSet

    def get_queryset(self):
        """
        Limit users to their own Tokens.
        """
        queryset = super().get_queryset()
        # Workaround for schema generation (drf_yasg)
        if getattr(self, "swagger_fake_view", False):
            return queryset.none()
        return queryset.filter(user=self.request.user)

    @action(
        detail=False,
        methods=["post"],
        authentication_classes=[authentication.TokenProvisionAuthentication],
        permission_classes=[],
    )
    def provision(self, request):
        """
        Non-authenticated REST API endpoint via which a user may create a Token.
        """
        # Create a new Token for the User
        token = Token.objects.create(user=request.user)
        data = serializers.TokenSerializer(token, context={"request": request}).data

        return Response(data, status=HTTP_201_CREATED)
```